static/js/components/forms/CheckoutForm.js

```diff
@@ -250,7 +250,12 @@ export class InnerCheckoutForm extends React.Component<InnerProps, InnerState> {
                     Apply
                   </button>
                 </div>
-                {formatErrors(errors.coupons)}
+                {!errors.coupons &&
+                values.couponCode &&
+                this.props.coupon &&
+                this.props.coupon.code === values.couponCode
+                  ? formatMessages("Success! Promo Code applied.")
+                  : formatErrors(errors.coupons)}
```

Comment:
This could use a helper function

```javascript
  isPromoCodeApplied = () => {
    const { coupon, values, errors } = this.props
      return (
        !errors.coupons &&
        values.couponCode &&
        coupon &&
        coupon.code === values.couponCode
    )

  render() {
    // ...
    { !this.isPromoCodeApplied()
        ? formatSuccessMessage("Success! Promo Code applied.")
        : formatErrors(errors.coupons)}
```